hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/streams/EventualAssertionResult.java

```diff
@@ -24,36 +24,72 @@
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 
+/**
+ * Represents the possibly pending result of an assertion. The result will be pending for at most the given timeout
+ * duration, after which the result will be completed as either a success or a failure based on the value of the
+ * {@code hasPassedIfNothingFailed} flag.
+ */
 public class EventualAssertionResult {
-    private final boolean hasPassedIfNothingFailed;
+    /**
+     * The maximum time the result can be pending before timing out.
+     */
     private final Duration timeout;
-
+    /**
+     * Whether the result on timing out should be taken as success.
+     */
+    private final boolean passAfterTimeout;
+    /**
+     * The latch that will be counted down when the result is ready.
+     */
     private final CountDownLatch ready = new CountDownLatch(1);
 
+    /**
+     * The result of the assertion.
+     */
     private AssertionResult result;
 
-    public EventualAssertionResult(final boolean hasPassedIfNothingFailed, @NonNull final Duration timeout) {
-        this.hasPassedIfNothingFailed = hasPassedIfNothingFailed;
+    /**
+     * Creates a new {@link EventualAssertionResult} with the given timeout and the given flag.
+     * @param passAfterTimeout whether the result on timing out should be taken as success
+     * @param timeout the maximum time the result can be pending before timing out
+     */
+    public EventualAssertionResult(final boolean passAfterTimeout, @NonNull final Duration timeout) {
+        this.passAfterTimeout = passAfterTimeout;
         this.timeout = requireNonNull(timeout);
     }
 
+    /**
+     * Blocks until the result is ready, or until the timeout has elapsed. If the result is not ready by the time the
+     * timeout has elapsed, the result will be completed as either a success or a timeout based on the value of the
+     * {@code passAfterTimeout} flag.
+     * @return the result of the assertion
+     * @throws InterruptedException if the thread is interrupted while waiting
+     */
     public AssertionResult get() throws InterruptedException {
         if (!ready.await(timeout.toMillis(), TimeUnit.MILLISECONDS)) {
-            if (hasPassedIfNothingFailed && result == null) {
-                return AssertionResult.success();
+            if (passAfterTimeout && result == null) {
+                return AssertionResult.newSuccess();
             } else {
-                return AssertionResult.timeout(timeout);
+                return AssertionResult.newTimeout(timeout);
```

Comment:
you can get here even if `result != null`, that doesn't seem right.   (There's a race where `pass()` could be called after the timeout before you get here; I think it is possible.)

People disparage `synchronized` methods and blocks as if they're _always_ bad and _always_ impact performance and affect concurrency, but on a very-low rate non-real-time critical class like this, why not use it?

Reply:
True, it's conceivable that `pass()` could be called just after the timeout period expires, but these timeouts are always set very liberally so won't be an issue in practice.

We can't make this method and the `fail()`/`pass()` methods all synchronized because then we couldn't call `pass()` once we started waiting.

Reply:
oh nuts you're right about synchronized.  And darn it that doesn't even rise to the level of "concurrency is hard".   So NVM that!